### PR TITLE
Added NULL checks in libpd.c

### PR DESCRIPTION
**Pull request by shilpa24balaji**

### CODE REVIEW

File: source/app/libpd.c [hunk 1]
```diff
@@ -35,62 +35,65 @@ static void connect_parodus()
...
-                        	backoffRetryTime = 0;
-                        	WalPrint("backoffRetryTime reached max value, reseting to initial value\n");
-                        }
-                }
-                retval = libparodus_shutdown(&current_instance);
-                WalPrint("libparodus_shutdown retval %d\n", retval);
-        }
+		
+		if(parodus_url != NULL && client_url != NULL)
+		{
```
Comment (selvamKrish) [inline source/app/libpd.c]: I guess we need to free parodus_url and client_url once we break out of while loop

File: source/app/libpd.c [hunk 2]
```diff
@@ -135,9 +139,15 @@ static void parodus_receive()
                                 res_wrp_msg->u.req.payload_size = strlen(res_wrp_msg->u.req.payload);
```
Comment (selvamKrish) [inline source/app/libpd.c]: Move WalPrint("Response payload is %s\n",(char *)(res_wrp_msg->u.req.payload)); inside NULL check

File: source/app/libpd.c (hunk 1 above)
Comment (sadhyama) [reply source/app/libpd.c]: After libparodus_init() we cannot free parodus_url and client_url as client reconnection is handled internally in libparodus which needs this urls later.

File: source/app/libpd.c (hunk 2 above)
Comment (sadhyama) [reply source/app/libpd.c]: Incorporated this change and amended the review.

**Pull request merged**